### PR TITLE
Change nullability annotation of HostString.Value to explictly allow null

### LINKED ISSUE #53567: Change nullability annotation of HostString.Value to explictly allow nulls
## Background and Motivation

It seems that HostString is intentionally implemented to handle null being passed to the constructor `HostString(string value)`, but currently the nullability annotation of this constructor does not allow null. I propose to explicitly allow a null value.

## Proposed API

```diff
namespace Microsoft.AspNetCore.Http;

public static class HostString
{
-    public HostString(string value)
+    public HostString(string? value)

-    public string Value
+    public string? Value
}
```

## Usage Examples

```csharp
var hostString = new HostString(null);
Console.WriteLine(hostString.Host); //outputs "" (empty string)
```

## Alternative Designs

Leave it as is

## Risks

I don't see any.

### PR DESCRIPTION
**Pull request by sorensenmatias**
# Change nullability annotation of HostString.Value to explictly allow null

- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Change nullability annotation of HostString.Value to explictly allow null.

## Description

Fixes #53567

### CODE REVIEW

File: src/Http/Http.Abstractions/src/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -1 +1,3 @@
 #nullable enable
+Microsoft.AspNetCore.Http.HostString.HostString(string? value) -> void
```
[the remark this answers is not in the document]
Comment (sorensenmatias) [reply src/Http/Http.Abstractions/src/PublicAPI.Unshipped.txt]: Thanks - fixed. I was not sure regarding the process for this (I could not find it in the documentation).

**Comment by martincostello** [comment]
I suggest you wait for #53567 to be approved before investing more time in this change.

**Comment by captainsafia** [comment]
> I suggest you wait for #53567 to be approved before investing more time in this change.

I commented on the other issue. We don't usually take nullability changes through API review. I reviewed the code here and I think this change is sensible.. 👍🏽

File: src/Http/Http.Abstractions/src/PublicAPI.Shipped.txt [hunk 2]
```diff
@@ -274,10 +274,10 @@ Microsoft.AspNetCore.Http.HostString.HasValue.get -> bool
 Microsoft.AspNetCore.Http.HostString.Host.get -> string!
 Microsoft.AspNetCore.Http.HostString.HostString() -> void
 Microsoft.AspNetCore.Http.HostString.HostString(string! host, int port) -> void
-Microsoft.AspNetCore.Http.HostString.HostString(string! value) -> void
```
[the remark this answers is not in the document]
Comment (sorensenmatias) [reply src/Http/Http.Abstractions/src/PublicAPI.Shipped.txt]: Thanks - fixed.

**Review by Jelly65 — APPROVED** [review]

**Review by captainsafia — APPROVED** [review]

**Pull request merged**